src/main/java/com/jumani/rutaseg/controller/OrderController.java

```diff
@@ -174,4 +178,60 @@ private OrderResponse createOrderResponse(Order order) {
                 customsDataResponse
         );
     }
+
+    @GetMapping("/search")
+    public ResponseEntity<List<OrderResponse>> search(
+            @RequestParam(value = "pema", required = false) Boolean pema,
+            @RequestParam(value = "transport", required = false) Boolean transport,
+            @RequestParam(value = "port", required = false) Boolean port,
+            @RequestParam(value = "arrivalDateFrom", required = false) LocalDate arrivalDateFrom,
+            @RequestParam(value = "arrivalDateTo", required = false) LocalDate arrivalDateTo,
+            @RequestParam(value = "arrivalTimeFrom", required = false) LocalTime arrivalTimeFrom,
+            @RequestParam(value = "arrivalTimeTo", required = false) LocalTime arrivalTimeTo,
+            @RequestParam(value = "clientId", required = false) Long clientId,
+            @RequestParam(value = "status", required = false) OrderStatus status,
+            @RequestParam(value = "pageSize", required = false, defaultValue = "10") Integer pageSize,
+            @Session SessionInfo session
+    ) {
+        final Long theClientId;
+        if (!session.admin()) {
+            theClientId = session.id();
+        } else {
+            theClientId = clientId;
+        }
+        final int thePageSize;
+        if (!session.admin()) {
+            thePageSize = 1;
+        } else {
+            thePageSize = pageSize;
+        }
+
+        if (Objects.nonNull(arrivalDateFrom) && Objects.nonNull(arrivalDateTo) && arrivalDateFrom.isAfter(arrivalDateTo)) {
+            throw new ValidationException("invalid_date_range", "the 'arrivalDateFrom' cannot be after 'arrivalDateTo'");
+        }
+
+        if (Objects.nonNull(arrivalTimeFrom) && Objects.nonNull(arrivalTimeTo)
+                && arrivalDateFrom.equals(arrivalDateTo) && arrivalTimeFrom.isAfter(arrivalTimeTo)) {
```

Comment:
ojo que en la 214 estás comparando equality de las dates jeje, también debería seguir la lógica del comentario anterior, creo que estaría bueno que permitamos mandar la misma hora para from y to

Reply:
deberiamos borrar el `&& arrivalDateFrom.equals(arrivalDateTo)` así permitimos que usen el mismo horario

Reply:
nico pero esto ya lo modifique, como decis que lo haga sino?